From f8438c46b25df9de7a1981d7778067046ce4ee1c Mon Sep 17 00:00:00 2001
From: Ning Feiyi <feiyix.ning@intel.com>
Date: Tue, 16 Jul 2013 10:34:03 +0800
Subject: [PATCH 234/429] mmc: SDHCI: The driver should not put the device into
 a low-power state

It is not required (in fact it even is not recommended) that a PCI driver's
suspend() callback save the standard configuration registers of the device,
prepare it for waking up the system, or put it into a low-power state.
All of these operations can very well be taken care of by the PCI subsystem,
without the driver's participation.

Signed-off-by: Ning Feiyi <feiyix.ning@intel.com>
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/host/sdhci-pci.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/drivers/mmc/host/sdhci-pci.c b/drivers/mmc/host/sdhci-pci.c
index fd14302fd314..d8aa4ba5c5fd 100644
--- a/drivers/mmc/host/sdhci-pci.c
+++ b/drivers/mmc/host/sdhci-pci.c
@@ -1723,19 +1723,6 @@ static int sdhci_pci_suspend(struct device *dev)
 			goto err_pci_suspend;
 	}
 
-	pci_save_state(pdev);
-	if (pm_flags & MMC_PM_KEEP_POWER) {
-		if (pm_flags & MMC_PM_WAKE_SDIO_IRQ) {
-			pci_pme_active(pdev, true);
-			pci_enable_wake(pdev, PCI_D3hot, 1);
-		}
-		pci_set_power_state(pdev, PCI_D3hot);
-	} else {
-		pci_enable_wake(pdev, PCI_D3hot, 0);
-		pci_disable_device(pdev);
-		pci_set_power_state(pdev, PCI_D3hot);
-	}
-
 	return 0;
 
 err_pci_suspend:
@@ -1755,12 +1742,6 @@ static int sdhci_pci_resume(struct device *dev)
 	if (!chip)
 		return 0;
 
-	pci_set_power_state(pdev, PCI_D0);
-	pci_restore_state(pdev);
-	ret = pci_enable_device(pdev);
-	if (ret)
-		return ret;
-
 	if (chip->fixes && chip->fixes->resume) {
 		ret = chip->fixes->resume(chip);
 		if (ret)
-- 
2.37.3

